### PR TITLE
Clang compiler

### PR DESCRIPTION
**Pull request by stigrj**
- Update MRCPP and XCFun to Clang friendly versions
- Adapt to minor changes in MRCPP interface
- Add Clang build on Travis

Status
--------
- [x] Ready to review
- [x] Ready to merge

### CODE REVIEW

**Comment by stigrj** [comment]
MRChem is actually warning free even with `-Weverything`. However, XCFun and MRCPP generates s**t loads of warnings, and fills the 10000 line limit in the Travis terminal output...

**Comment by robertodr** [comment]
:+1: for a warning-free MRChem. Less so for the rest. Some warning in XCFun are inevitable, if I recall correctly, due to the liberal use of preprocessor macros to generate code. For MRCPP, could be a nice 1-hour programming exercise in Haraldvollen? "Whack-a-warning"?

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "external/xcfun"]
 	path = external/xcfun
-	url = https://github.com/dftlibs/xcfun.git
+	url = https://github.com/stigrj/xcfun.git
```
Comment (robertodr) [inline .gitmodules]: Your latest PR has been merged: this one can point back to the canonical DFTlibs repo.

File: src/utils/gto_utils/AOContraction.cpp [hunk 2]
```diff
@@ -103,10 +103,11 @@ GaussExp<3> AOContraction::getContraction(int m, const mrcpp::Coord<3> &center)
 
     normFac = std::sqrt(normFac);
     for (unsigned int i = 0; i < expo.size(); i++) {
-        GaussFunc<3> gto;
-        gto.setPos(center.data());
-        gto.setExp(this->expo[i]);
-        gto.setPower(GTOS[this->L].comp[m]);
+        std::array<int, 3> pow;
+        pow[0] = GTOS[this->L].comp[m][0];
```
Comment (robertodr) [inline src/utils/gto_utils/AOContraction.cpp]: Nit-pick: `std::copy_n(pow.begin(), 3, GTO[this->L].comp[m].begin());` It also seems `pow` can be declared and filled outside the loop over?

**Pull request merged**